src/tickets/tickets.service.ts

```diff
@@ -0,0 +1,95 @@
+import { httpErrors } from '@fastify/sensible'
+import { sql } from 'drizzle-orm'
+import { FastifyInstance, FastifyRequest } from 'fastify'
+import { TCreateTicketSchema } from './schema/create-ticket.schema'
+import { getUserBySessionId } from '../auth/auth.service'
+import { TUpdateTicketSchema } from './schema/update-ticket.schema'
+
+export const getTicketsHandler = async (app: FastifyInstance) => {
+  const response = app.db.query.tickets.findMany({
+    with: { attachments: true },
+  })
+
+  return response
+}
+
+export const getTicketHandler = async (app: FastifyInstance, uuid: string) => {
+  const response = await app.db.execute(sql`
+    SELECT * FROM tickets
+    WHERE uuid = ${uuid}
+    `)
+
+  if (!response.rows.length) {
+    throw httpErrors.notFound('Ticket not found')
+  }
+
+  return response.rows
+}
+
+export const postTicketHandler = async (
+  app: FastifyInstance,
+  request: FastifyRequest
+) => {
+  const sessionId = request.session?.sessionId
+  const user = await getUserBySessionId(app.db, sessionId)
+
+  if (!user) {
+    throw httpErrors.notFound('User not found')
+  }
+
+  const ticketData = request.body as TCreateTicketSchema
+
+  const result = await app.db.execute(sql`
+      INSERT INTO tickets (title, description, creator_uuid)
+      VALUES (${ticketData.title}, ${ticketData.description}, ${user.user_uuid})
+      RETURNING *
+    `)
+
+  return result.rows
+}
+
+export const patchTicketHandler = async (
+  app: FastifyInstance,
+  request: FastifyRequest
+) => {
```

Comment:
I think overall in services I'd avoid passing whole apps and requests. you can have db, and then necessary data

Reply:
you don't have to change it now, just for the future

Reply:
I agree 👍